xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -249,9 +249,13 @@ void CGUIDialogVideoInfo::OnInitWindow()
 
   const std::shared_ptr<CProfileManager> profileManager = CServiceBroker::GetSettingsComponent()->GetProfileManager();
 
-  CONTROL_ENABLE_ON_CONDITION(CONTROL_BTN_REFRESH,
-      (profileManager->GetCurrentProfile().canWriteDatabases() || g_passwordManager.bMasterUser) &&
-      !StringUtils::StartsWithNoCase(m_movieItem->GetProperty("xxuniqueid").c_str(), "xx"));
+  if (m_movieItem->GetProperty("xxuniqueid").empty() ||
+              !StringUtils::StartsWithNoCase(m_movieItem->GetProperty("xxuniqueid").c_str(), "xx"))
+    CONTROL_ENABLE_ON_CONDITION(CONTROL_BTN_REFRESH,
+                                (profileManager->GetCurrentProfile().canWriteDatabases() ||
+                                g_passwordManager.bMasterUser));
+  else
+    CONTROL_DISABLE(CONTROL_BTN_REFRESH);
```

Comment:
```c++
  const std::string uniqueId = m_movieItem->GetProperty("xxuniqueid"); 
  if (uniqueId.empty() || !StringUtils::StartsWithNoCase(uniqueId.c_str(), "xx"))
    CONTROL_ENABLE_ON_CONDITION(CONTROL_BTN_REFRESH, (profileManager->GetCurrentProfile().canWriteDatabases() || g_passwordManager.bMasterUser));
  else
    CONTROL_DISABLE(CONTROL_BTN_REFRESH);
```

... to avoid calling rather expensive `GetProperty()` twice.

Reply:
@ksooo thanx for the tip, but that doesn't compile..

```
/home/ronie/software/kodi/xbmc/video/dialogs/GUIDialogVideoInfo.cpp:253:56: error: conversion from ‘const CVariant’ to non-scalar type ‘const string’ {aka ‘const std::__cxx11::basic_string<char>’} requested
  253 |   const std::string uniqueId = m_movieItem->GetProperty("xxuniqueid");
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
make[2]: *** [build/video/dialogs/CMakeFiles/video_dialogs.dir/build.make:160: build/video/dialogs/CMakeFiles/video_dialogs.dir/GUIDialogVideoInfo.cpp.o] Error 1
```

Reply:
`GetProperty(...).asString();` - sorry.

Reply:
```diff
diff --git a/xbmc/video/dialogs/GUIDialogVideoInfo.cpp b/xbmc/video/dialogs/GUIDialogVideoInfo.cpp
index ea5a9fcadc..96191a6583 100644
--- a/xbmc/video/dialogs/GUIDialogVideoInfo.cpp
+++ b/xbmc/video/dialogs/GUIDialogVideoInfo.cpp
@@ -249,9 +249,14 @@ void CGUIDialogVideoInfo::OnInitWindow()
 
   const std::shared_ptr<CProfileManager> profileManager = CServiceBroker::GetSettingsComponent()->GetProfileManager();
 
-  CONTROL_ENABLE_ON_CONDITION(CONTROL_BTN_REFRESH,
-      (profileManager->GetCurrentProfile().canWriteDatabases() || g_passwordManager.bMasterUser) &&
-      !StringUtils::StartsWithNoCase(m_movieItem->GetProperty("xxuniqueid").c_str(), "xx"));
+  const std::string uniqueId = m_movieItem->GetProperty("xxuniqueid").asString();
+  if (uniqueId.empty() || !StringUtils::StartsWithNoCase(uniqueId.c_str(), "xx"))
+    CONTROL_ENABLE_ON_CONDITION(CONTROL_BTN_REFRESH,
+                                (profileManager->GetCurrentProfile().canWriteDatabases() ||
+                                 g_passwordManager.bMasterUser));
+  else
+    CONTROL_DISABLE(CONTROL_BTN_REFRESH);
+
   CONTROL_ENABLE_ON_CONDITION(CONTROL_BTN_GET_THUMB,
       (profileManager->GetCurrentProfile().canWriteDatabases() || g_passwordManager.bMasterUser) &&
       !StringUtils::StartsWithNoCase(m_movieItem->GetVideoInfoTag()->
```

Reply:
thanx! compiles and runs fine now.